pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -329,12 +328,31 @@ func (plugin *kubenetNetworkPlugin) Capabilities() utilsets.Int {
 	return utilsets.NewInt()
 }
 
-// setup sets up networking through CNI using the given ns/name and sandbox ID.
-func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kubecontainer.ContainerID, annotations map[string]string) error {
-	var ipv4, ipv6 net.IP
+// adds IP addresses to the plugin and return the Pods gateway and cidr
+func (plugin *kubenetNetworkPlugin) trackPodIPs(id kubecontainer.ContainerID, res *cnitypescurrent.Result) ([]net.IP, []net.IPNet) {
 	var podGateways []net.IP
 	var podCIDRs []net.IPNet
 
+	// Logic taken from: https://github.com/kubernetes/kubernetes/pull/69821
+	for _, ip := range res.IPs {
+		if ip.Interface == nil || *ip.Interface > len(res.Interfaces) {
+			continue
+		}
+
+		intf := res.Interfaces[*ip.Interface]
+		if intf.Sandbox != "" && intf.Name == network.DefaultInterfaceName {
+			// Found our sandbox interface, use the IP
+			podGateways = append(podGateways, ip.Gateway)
+			podCIDRs = append(podCIDRs, net.IPNet{IP: ip.Address.IP.Mask(ip.Address.Mask), Mask: ip.Address.Mask})
+			plugin.addPodIP(id, ip.Address.IP.String())
+		}
+	}
+
+	return podGateways, podCIDRs
```

Comment:
That will blank out PodIPs. Is this what we want?

Reply:
The `plugin.addPodIP` is called directly in the for loop instead afterwards so the behaviour should be the same?